drivers/adc/adc_mcux_adc12.c

```diff
@@ -253,6 +264,9 @@ static const struct adc_driver_api mcux_adc12_driver_api = {
 #ifdef CONFIG_ADC_ASYNC
 	.read_async = mcux_adc12_read_async,
 #endif
+#if defined(CONFIG_SOC_SERIES_S32K1)
+	.ref_internal = ADC_S32K1_REF_INTERNAL_MV,
```

Comment:
I propose to do this in a more generic way by defining a dt prop as other bindings do, e.g.
```
  vref-mv:
    type: int
    description: Indicates the reference voltage of the ADC in mV.
```
and then set it in the `ucans32k1sic` board's overlay as `vref-mv: <5000>;`. It is better to keep the drivers decoupled as much as possible from SoC or board specific values.

Reply:
Good point. However, it appears that the adc_mcux_adc12 driver only supports reference = ADC_REF_INTERNAL. This means that the conversion function (adc_raw_to_millivolts_dt) uses ref_internal instead of vref_mv.

Reply:
that's correct, the value of `vref-mv` can be used to set `api.ref_internal`, see for instance `adc_npcx.c`, `adc_nxp_s32_adc_sar.c` or `adc_rpi_pico.c` for inpiration